sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -58,6 +59,33 @@ export class EventHubProducer {
     this._eventHubSender = EventHubSender.create(this._context, partitionId);
   }
 
+  /**
+   * Creates a batch where event data objects can be added for later `Send` call
+   * @param options  Options to define partition key and max message size.
+   * @returns Promise<EventDataBatch>
+   */
+  async createBatch(options?: BatchOptions): Promise<EventDataBatch> {
+    if (!options) {
+      options = {};
+    }
+    let maxMessageSize = await this._eventHubSender!.getMaxMessageSize();
+    if (options.maxMessageSize) {
+      if (options.maxMessageSize > maxMessageSize) {
+        const error = new Error(
+          "Max message size is greater than maximum message size on the AMQP sender link."
```

Comment:
Nice!

I think it would also be helpful to add the maximum message size reported by the AMQP link in this error and the logs.